src/session/error.rs

```diff
@@ -179,3 +185,42 @@ impl From<MissingFieldError> for ParseError {
         ParseError::malformed(error)
     }
 }
+
+/// An error type for situations where proto enum has a value for which the definition is not known.
+#[derive(Debug)]
+pub struct UnknownEnumValueError {
+    pub enum_name: &'static str,
+    pub value: i32
+}
+
+impl UnknownEnumValueError {
+
+    /// Creates a new error indicating that a proto enum has a value for which the definition
+    /// is not known.
+    pub fn new(enum_name: &'static str, value: i32) -> UnknownEnumValueError {
```

Comment:
In general, I don't think this constructor is useful. I think initializing this error always with the record syntax has better readability.

Reply:
Removed.